src/renderer/components/opds/BrowserResult.tsx

```diff
@@ -76,38 +88,46 @@ export class BrowserResult extends React.Component<BrowserResultProps, null> {
                     default:
                         break;
                 }
-            } else if (result.isTimeout) {
+            } else if (browserResult.isTimeout) {
                 content = (
                     <MessageOpdBrowserResult title={__("opds.network.timeout")} />
                 );
             } else {
                 content = (
                     <MessageOpdBrowserResult
                         title={__("opds.network.error")}
-                        message={`${result.statusCode || "unknow error code"} : ${result.statusMessage || ""}`}
+                        message={`${browserResult.statusCode || "unknow error code"} : ${browserResult.statusMessage || ""}`}
                     />
                 );
             }
         }
 
         return <div className={styles.opdsBrowseContent}>
-            { content }
+            {content}
         </div>;
     }
 
     private browseOpds() {
-        const { url, location, result, browse } = this.props;
+        const { url, location } = this.props;
+        const { browserResult } = this.state;
         const oldQs = parseQueryString(url.split("?")[1]);
         const search = qs.parse(location.search.replace("?", "")).search;
         let newUrl = url;
-        if (search && result && typeof result === "object" && result.isSuccess && result.data.searchUrl) {
-            newUrl = result.data.searchUrl;
+
+        if (search && browserResult && browserResult.isSuccess && browserResult.data.searchUrl) {
+            newUrl = browserResult.data.searchUrl;
             newUrl = this.addSearchTerms(newUrl, search) +
                 Object.keys(oldQs).map((id) => `&${id}=${oldQs[id]}`).join("");
         }
+
         this.currentUrl = newUrl;
-        this.props.cleanData();
```

Comment:
See my comment about "clean":
https://github.com/readium/readium-desktop/pull/696/files#r327787727

Reply:
It's automatically handle with apiFetch